tests/integration/targets/aci_epg_useg_attribute/tasks/mac_attr.yml

```diff
@@ -0,0 +1,191 @@
+# Test code for the ACI modules
+# Copyright: (c) 2023, Christian Kolrep (@Christian-Kolrep)
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: Test that we have an ACI APIC host, ACI username and ACI password
+  fail:
+    msg: 'Please define the following variables: aci_hostname, aci_username and aci_password.'
+  when: aci_hostname is not defined or aci_username is not defined or aci_password is not defined
+
+# CLEAN ENVIRONMENT
+- name: Set vars
+  set_fact:
+    aci_info: &aci_info
+      host: "{{ aci_hostname }}"
+      username: "{{ aci_username }}"
+      password: "{{ aci_password }}"
+      validate_certs: '{{ aci_validate_certs | default(false) }}'
+      use_ssl: '{{ aci_use_ssl | default(true) }}'
+      use_proxy: '{{ aci_use_proxy | default(true) }}'
+      output_level: '{{ aci_output_level | default("debug") }}'
+
+- name: Query system information
+  aci_system:
+    <<: *aci_info
+    id: 1
+    state: query
+  register: version
+
+- name: Verify Cloud and Non-Cloud Sites in use.
+  include_tasks: ../../../../../../integration/targets/aci_cloud_provider/tasks/main.yml
+
+- name: Execute tasks only for non-cloud sites
+  when: query_cloud.current == []  # This condition will execute only non-cloud sites
+  block:  # block specifies execution of tasks within, based on conditions
+  - name: ensure tenant exists for tests to kick off
+    cisco.aci.aci_tenant: &aci_tenant_present
+      <<: *aci_info
+      tenant: ansible_test
+      state: present
+    register: tenant_present
+
+  - name: ensure ap exists for tests to kick off
+    cisco.aci.aci_ap: &aci_ap_present
+      <<: *aci_tenant_present
+      ap: anstest
+    register: ap_present
+
+  - name: ensure bd exists for tests to kick off
+    cisco.aci.aci_bd: &aci_bd_present
+      <<: *aci_tenant_present
+      bd: anstest
+    register: bd_present
+
+  - name: ensure useg epg exists for tests to kick off
+    cisco.aci.aci_epg: &aci_epg_present
+      <<: *aci_ap_present
+      epg: anstest
+      bd: anstest
+      useg: yes
+    register: epg_present
+
+  - name: ensure useg attribute of type mac not exists for tests to kick off
+    cisco.aci.aci_epg_useg_attribute:
+      <<: *aci_ap_present
+      epg: anstest
+      useg_attribute_name: anstest
+      useg_attribute_type: mac
+      state: absent
+
+  - name: Add a useg attribute of type mac ( check mode )
+    cisco.aci.aci_epg_useg_attribute: &aci_mac_present
+      <<: *aci_ap_present
+      epg: anstest
+      useg_attribute_name: anstest
+      useg_attribute_type: mac
+      value: "aa:bb:cc:dd:ee:ff"
+    check_mode: true
+    register: mac_present_check_mode
+
+  - name: Add a useg attribute of type mac - creation works
+    cisco.aci.aci_epg_useg_attribute:
+      <<: *aci_mac_present
+    register: mac_present
```

Comment:
The task failed due to the DN not found error message.

Reply:
seems like removing name="default" in criterion config prevented its creation. after reverting tests pass.